source/common/stats/utility.cc

```diff
@@ -34,5 +35,84 @@ absl::optional<StatName> Utility::findTag(const Metric& metric, StatName find_ta
   return value;
 }
 
+namespace {
+
+// Helper class for the three Utility::*FromElements implementations to build up
+// a joined StatName from a mix of StatName and string_view.
+struct ElementVisitor {
+  ElementVisitor(SymbolTable& symbol_table) : symbol_table_(symbol_table), pool_(symbol_table) {}
+
+  // Overloads provides for absl::visit to call.
+  void operator()(StatName stat_name) { stat_names_.push_back(stat_name); }
+  void operator()(absl::string_view name) { stat_names_.push_back(pool_.add(name)); }
+
+  // Generates a StatName from the elements.
+  StatName makeStatName(const ElementVec& elements) {
```

Comment:
maybe add a comment saying this shouldn't be called more than once, since it seems like we'd concatenate it with the previous result? Or perhaps this should all be done in the constructor to avoid this oddity completely?

Reply:
I like the idea of doing the work in the constructor here. Much nicer.